pkg/workspace/library_loader.go

```diff
@@ -0,0 +1,190 @@
+package workspace
+
+import (
+	"fmt"
+
+	"github.com/k14s/ytt/pkg/files"
+	"github.com/k14s/ytt/pkg/texttemplate"
+	"github.com/k14s/ytt/pkg/yamlmeta"
+	"github.com/k14s/ytt/pkg/yamltemplate"
+	"github.com/k14s/ytt/pkg/yttlibrary"
+)
+
+type LibraryLoader struct {
+	*Library
+	UI     files.UI
+	Opts   TemplateLoaderOpts
+	Values interface{}
+}
+
+type EvalResult struct {
+	DocSet      *yamlmeta.DocumentSet
+	DocSets     map[*FileInLibrary]*yamlmeta.DocumentSet
+	OutputFiles []files.OutputFile
+}
+
+func NewLibraryLoader(lib *Library, parentValuesAst interface{}, ui files.UI, loaderOpts TemplateLoaderOpts) *LibraryLoader {
+	return &LibraryLoader{
+		UI:      ui,
+		Opts:    loaderOpts,
+		Library: lib,
+		Values:  parentValuesAst,
+	}
+}
+
+func (ll *LibraryLoader) LoadValues() error {
```

Comment:
let's also combine LoadValues() and Eval()

Reply:
I did not combine them because the `--data-values-inspect` flag needs to hook after data values processing but must not evaluate templates.

Reply:
i see. we should then change LoadValues() to Values() which returns them and then can be passed into Eval(), which would make this object not mutable (ll.Values fields).

Reply:
Ok, I'm changing the LoadValues function to take parent values override and return the values from the value file with the parent override. Eval() will now take a value parameter.